runtime-modules/working-team/src/benchmarking.rs

```diff
@@ -0,0 +1,921 @@
+    }
+
+    apply_on_opening {
+      let i in 1 .. 50000;
+
+      let (lead_account_id, lead_member_id) = member_funded_account::<T>("lead", 0);
+      let opening_id = add_opening_helper::<T, I>(0, &T::Origin::from(RawOrigin::Root), &StakingRole::WithStakes, &JobOpeningType::Leader);
+
+      let apply_on_opening_params = ApplyOnOpeningParameters::<T, I> {
+        member_id: lead_member_id,
+        opening_id: opening_id.clone(),
+        role_account_id: lead_account_id.clone(),
+        reward_account_id: lead_account_id.clone(),
+        description: vec![0u8; i.try_into().unwrap()],
+        stake_parameters: Some(
+          // Make sure to keep consistency with the StakePolicy in add_opening_helper (we are safe as long as we are
+          // using max_value for stake)
+          StakeParameters {
+            stake: One::one(),
+            staking_account_id: lead_account_id.clone(),
+          }
+        ),
+      };
+
+    }: _ (RawOrigin::Signed(lead_account_id.clone()), apply_on_opening_params)
+    verify {
+      assert!(ApplicationById::<T, I>::contains_key(T::ApplicationId::from(0)), "Application not found");
+      assert_last_event::<T, I>(RawEvent::AppliedOnOpening(opening_id, Zero::zero()).into());
+    }
+
+    fill_opening_lead {
+      let i in 0 .. 10;
+
+      let (lead_account_id, lead_member_id) = member_funded_account::<T>("lead", 0);
+      let (opening_id, application_id) = add_and_apply_opening::<T, I>(0, &RawOrigin::Root.into(), &StakingRole::WithoutStakes, &lead_account_id,
+        &lead_member_id, &JobOpeningType::Leader);
+
+      let mut successful_application_ids: BTreeSet<T::ApplicationId> = BTreeSet::new();
+      successful_application_ids.insert(application_id);
+    }: fill_opening(RawOrigin::Root, opening_id, successful_application_ids)
+    verify {
+      assert!(!OpeningById::<T, I>::contains_key(opening_id), "Opening still not filled");
+      assert_eq!(WorkingTeam::<T, I>::current_lead(), Some(Zero::zero()), "Opening for lead not filled");
+    }
+
+    fill_opening_worker { // We can actually fill an opening with 0 applications?
+      let i in 1 .. T::MaxWorkerNumberLimit::get();
+      let (lead_id, lead_worker_id) = insert_a_worker::<T, I>(StakingRole::WithoutStakes, JobOpeningType::Leader, 0, None);
+
+      let (opening_id, successful_application_ids, _) = add_opening_and_n_apply::<T, I>(
+        &(1..i).collect(),
+        &T::Origin::from(RawOrigin::Signed(lead_id.clone())),
+        &StakingRole::WithoutStakes,
+        &JobOpeningType::Regular
+      );
+    }: fill_opening(RawOrigin::Signed(lead_id.clone()), opening_id, successful_application_ids.clone())
+    verify {
+      assert!(!OpeningById::<T, I>::contains_key(opening_id), "Opening still not filled");
+
+      for i in 1..successful_application_ids.len() {
+        assert!(WorkerById::<T, I>::contains_key(TeamWorkerId::<T>::from(i.try_into().unwrap())), "Not all workers
+        added");
+      }
```

Comment:
It seems that you missed check for the `OpeningFilled` event.

Reply:
Also, a successful outcome contains a new worker in the storage.

Reply:
> Also, a successful outcome contains a new worker in the storage.

Aren't we checking that the outcome contains a new worker in the storage by checking that:
```rs
WorkerById::<T, I>::contains_key(TeamWorkerId::<T>::from(i.try_into().unwrap()))
```

is true?

Reply:
Nevermind, probably I saw the previous commit version.